com.unity.netcode.gameobjects/Tests/Runtime/RpcManyClientsTests.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Netcode.RuntimeTests
+{
+    public class RpcManyClientsObject : NetworkBehaviour
+    {
+        public int Count = 0;
+        [ServerRpc(RequireOwnership = false)]
+        public void ResponseServerRpc()
+        {
+            Count++;
+        }
+
+        [ClientRpc]
+        public void SomeClientRpc()
+        {
+            ResponseServerRpc();
+        }
+        [ClientRpc]
+        public void SomeClientRpc(int value)
+        {
+            ResponseServerRpc();
+        }
+        [ClientRpc]
+        public void SomeClientRpc(int value1, int value2)
+        {
+            ResponseServerRpc();
+        }
+    }
+
+    public class RpcManyClientsTests : BaseMultiInstanceTest
+    {
+        protected override int NbClients => 10;
+
+        private GameObject m_PrefabToSpawn;
+
+        [UnitySetUp]
+        public override IEnumerator Setup()
+        {
+            yield return StartSomeClientsAndServerWithPlayers(useHost: true, nbClients: NbClients,
+                updatePlayerPrefab: playerPrefab =>
+                {
+                    m_PrefabToSpawn = PreparePrefab(typeof(RpcManyClientsObject));
+                });
+        }
+
+        public GameObject PreparePrefab(Type type)
```

Comment:
Did it cross your mind that maybe we could include this in our standard testing code / MultiInstanceHelpers?  I don't see anything "RpcMany..." in this.

Reply:
not a bad idea — I'd like to summon @NoelStephensUnity here since he's been looking into testing stuff recently.
I still believe it needs to be addressed under another PR though (this PR is just a backport)

Reply:
@mattwalsh-unity This is coming shortly in #1687. It looks like a monster, but really the bulk of it is namespace and class name changes (i.e. BaseMultiInstanceTest is now called NetcodeIntegrationTest) and file migrations (i.e. like all helper Unity.Netcode.RuntimeTests helper classes moved into a neutral assembly.

- The timing issues regarding waiting for a condition is now integrated into the NetcodeIntegrationTest class
- It also provides much more modularity where you can choose to have it create NetworkManagers for server and clients each test, to re-use the same instances for all tests, or to not create anything and let the developer handle all of that.
- The "new" wait for condition is "WaitForConditionOrTimeOut" which also provides developers the ability to assign their own TimeOutHelper in the event they have special needs.  It basically does what it sounds like... 

The PR should be done by later tomorrow or early Thursday... so some goodness is on its way.